### PR TITLE
option oversampling template

### PR DESCRIPTION
**Pull request by JanaKoehler1985**
Oversampling of the template file. 
Have put it outside the instrument file, so the user can chose if wanted or not and how much.

### CODE REVIEW

File: viper.py [hunk 1]
```diff
@@ -558,7 +560,14 @@ def fit_chunk(o, chunk, obsname, targ=None, tpltarg=None):
     print('reading stellar template')
     w_tpl, f_tpl = {}, {}
     for o in orders:
-        w_tpl[o], f_tpl[o] = Tpl(tplname, o=o, targ=targ)
+        w_tplo, f_tplo = Tpl(tplname, o=o, targ=targ)
+        if oversampling:
+            us = np.linspace(np.log(w_tplo[0]),np.log(w_tplo[-1]), oversampling*w_tplo.size)
+            f_tplo[np.isnan(f_tplo)] = 0
```
Comment (mzechmeister) [inline viper.py:566]: There is also the function `np.nan_to_num`.

**Comment by mzechmeister** [comment]
I see the need for such an option.
Now oversampling and cubic splines also enters `viper`. (`serval` has an option `ofac` and custom cubic spline regression).

Why do you use log flux in interpolation?
https://github.com/mzechmeister/viper/blob/6f3670d73ecc3fdd014b9681bc9487e966a20ab0/viper.py#L567
This can give `nan` and `-inf` fluxes.

Moreover, I think the splining should happen along with the conversion into a function:
https://github.com/mzechmeister/viper/blob/6f3670d73ecc3fdd014b9681bc9487e966a20ab0/viper.py#L223-L225
Actually, the `S_star` could take the form
```python
S_star = CubicSpline(np.log(w_tpl[o])-berv/c, f_tpl[o])
```
That would be best and would not need an oversampling parameter. But I'm afraid, that cubic splining in every model call is slow.
That's why the cubic spline model is discretised and finally linear splined.

Anyway, the cubic and linear spline should be both either inside `fit_chunk` or both outside `fit_chunk`. In the later case, we would only pass the stellar model (`S_star`, while currently `w_tpl` and `f_tpl` are a global variables :unamused:).

It calls for some major restructuring.

**Comment by JanaKoehler1985** [comment]
As far as I see it, I just use log wavelength and no log flux. Or is my brain already totally melted in the heat?

I' m not sure I like the idea of putting the CubicSpline function into the S_star definition. It would indeed lead to a lot of major restructuring in the viper.py and model.py files and I don't know if we do us a favor with it. Besides I don't see, why this would be needed or why it would improve things. Can you explain me, why it would be better that way?

I would prefer to leaf the Cubicspline where it is in the moment, but be fine with leaving the oversampling. Made a short test run on the CRIRES+ data and the difference between using and not using oversampling is not that large, Cubicspline performed in both cases  (with oversampling: RV, eRV, prms:  -333.36, 30.5, 2.75; no oversampling:  RV, eRV, prms:  -334.29, 34.06, 2.82).

**Comment by mzechmeister** [comment]
> As far as I see it, I just use log wavelength and no log flux. Or is my brain already totally melted in the heat?

No, I need new glasses. I was probably confused by the two `log`s in
https://github.com/mzechmeister/viper/blob/6f3670d73ecc3fdd014b9681bc9487e966a20ab0/viper.py#L565

> major restructuring ... Besides I don't see, why this would be needed or why it would improve things. Can you explain me, why it would be better that way?

The improvement will be only in code design, i.e. cleaner.
 
> I would prefer to leaf the Cubicspline where it is in the moment, but be fine with leaving the oversampling.

Ok.

**Pull request merged**